### DISCUSSION TITLE
Explain why the Javascript Adapter returns `true` instead of a meaningful errormessage when errors occur while updateing tokens

### DISCUSSION THREAD

**Discussion by ThomasJunk** [keycloak/keycloak#28920] [Q&A]
### Description

I wrote an application which contains the following code Block

```
try {
      await keycloak.updateToken();
      return true;
    } catch (error) {
      return false;
    }
```

In case of an error I wanted to leverage information from `error`. Unfortunately it seems it is just set to `true`. See https://github.com/keycloak/keycloak/blob/main/js/libs/keycloak-js/src/keycloak.js#L689

Is my analysis correct? And if so why isn't the error from the response retured?

**Comment by tekhedd** [comment]
I am seeing the same behavior. It could be a bug related to the old behavior, which called different functions for "refreshed" and "not refreshed". Perhaps just an oversight?
